modules/space/translation/horizonstranslation.cpp

```diff
@@ -130,13 +128,48 @@ glm::dvec3 HorizonsTranslation::position(const UpdateData& data) const {
     return interpolatedPos;
 }
 
-void HorizonsTranslation::readHorizonsTextFile(const std::string& horizonsTextFilePath) {
-    std::ifstream fileStream(horizonsTextFilePath);
+void HorizonsTranslation::loadData() {
+    std::string file = _horizonsTextFile;
+    if (!FileSys.fileExists(absPath(file))) {
+        return;
+    }
+
+    std::string cachedFile = FileSys.cacheManager()->cachedFilename(
+        file,
+        ghoul::filesystem::CacheManager::Persistent::Yes
+    );
+
+    bool hasCachedFile = FileSys.fileExists(cachedFile);
+    if (hasCachedFile) {
+        LINFO(fmt::format("Cached file '{}' used for Horizon file '{}'", cachedFile, file));
+
+        bool success = loadCachedFile(cachedFile);
+        if (success) {
+            return;
+        }
+        else {
+            FileSys.cacheManager()->removeCacheFile(file);
+            // Intentional fall-through to the 'else' computation to generate the cache
+            // file for the next run
+        }
+    }
+    else {
+        LINFO(fmt::format("Cache for Horizon file '{}' not found", file));
+    }
+    LINFO(fmt::format("Loading Horizon file '{}'", file));
+
+    readHorizonsTextFile();
+
+    LINFO("Saving cache");
+    saveCachedFile(cachedFile);
+}
+
+void HorizonsTranslation::readHorizonsTextFile() {
+    std::ifstream fileStream(_horizonsTextFile);
 
     if (!fileStream.good()) {
         LERROR(fmt::format(
-            "Failed to open Horizons text file '{}'", horizonsTextFilePath
-        ));
+            "Failed to open Horizons text file '{}'", _horizonsTextFile));
```

Comment:
Nitpicky:  the `));` would be on the next line, starting on the same column as `LERROR`